### PR TITLE
Boostrap server

### PR DESCRIPTION
**Pull request by edisonz0718**

### CODE REVIEW

File: api/cmd/main.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+			PrivateKey: privateKey,
+		})
+		if err != nil {
+			fmt.Printf("Failed to set up sdk: %v", err)
+			c.String(http.StatusInternalServerError, "Internal server error")
+			return
+		}
+
+		// You can replace your own contract address here
+		contractAddress := "0xbcF76d7B52D6edef3D6Ec009D0371F06895B4E6A"
```
Comment (kingster-will) [inline api/cmd/main.go:75]: The latest contract address is deployed at: 0x64432E5A76a93e79be2f7F3F12982059a32Fd794

**Comment by kingster-will** [comment]
LGTM!

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	docker-compose build $*
+
+push-api:
+	docker tag ${API_SERVER_IMAGE} ${ECR}/${API_SERVER_IMAGE}:${TAG}
+	docker tag ${API_SERVER_IMAGE} ${ECR}/${API_SERVER_IMAGE}:latest
+	docker push ${ECR}/${API_SERVER_IMAGE}:${TAG}
+	docker push ${ECR}/${API_SERVER_IMAGE}:latest	
+
+deploy-%:
+	cd $*; make deploy
```
Comment (LeoHChen) [inline Makefile:47]: minor: add a "make help" to provide some help message

File: api/cmd/main.go [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+			return
+		}
+
+		c.JSON(http.StatusOK, gin.H{
+			"proof": *result.MerkleProof,
+		})
+	})
+
+	r.GET("/nfts", func(c *gin.Context) {
+		privateKey := "a6cd3f393b1cddf8be66e2ff784640adbafbce852267ec1ec0e22eb741232000"
```
Comment (LeoHChen) [inline api/cmd/main.go:63]: best practice: never add any private key to the code. to simplify, use an external environment variable or config file.

File: api/cmd/main.go [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
...uts":[{"internalType":"contract IOffChainDataSource","name":"offChainDS_","type":"address"}],"name":"setOffChainDataSource","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"bytes4","name":"interfaceId","type":"bytes4"}],"name":"supportsInterface","outputs":[{"internalType":"bool","name":"","type":"bool"}],"stateMutability":"view","type":"function"},{"inputs":[],"name":"symbol","outputs":[{"internalType":"string","name":"","type":"string"}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"uint256","name":"","type":"uint256"}],"name":"tokenGrades","outputs":[{"internalType":"uint256","name":"","type":"uint256"}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"uint256","name":"tokenId_","type":"uint256"}],"name":"tokenURI","outputs":[{"internalType":"string","name":"","type":"string"}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"address","name":"from","type":"address"},{"internalType":"address","name":"to","type":"address"},{"internalType":"uint256","name":"tokenId","type":"uint256"}],"name":"transferFrom","outputs":[],"stateMutability":"nonpayable","type":"function"}]`
```
Comment (LeoHChen) [inline api/cmd/main.go:78]: read ABI from an external file.

File: pkg/utils/ethutils.go [hunk 5]
```diff
@@ -0,0 +1,66 @@
+package utils
```
Comment (LeoHChen) [inline pkg/utils/ethutils.go:1]: Is this code from somewhere else? please add a reference if yes.

File: api/cmd/main.go [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+	r.GET("/mint/proof", func(c *gin.Context) {
+		address := c.DefaultQuery("address", "")
+
+		// Validate address
+		if address == "" {
+			c.String(http.StatusBadRequest, fmt.Sprintf("input address is invalid, address: %s", address))
+			return
+		}
+
+		db, err := database.NewGormDB("postgresql://postgres:@localhost:23005/postgres?sslmode=disable")
```
Comment (LeoHChen) [inline api/cmd/main.go:44]: minor: use a config file

**Review by LeoHChen — CHANGES_REQUESTED** [review]
please don't hardcode any key in the file or PR submitted to github.

File: api/cmd/main.go (hunk 1 above)
Comment (edisonz0718) [reply api/cmd/main.go]: Updated. Thanks

File: api/cmd/main.go (hunk 4 above)
Comment (edisonz0718) [reply api/cmd/main.go]: Updated

File: pkg/utils/ethutils.go (hunk 5 above)
Comment (edisonz0718) [reply pkg/utils/ethutils.go:1]: Done

File: api/cmd/main.go (hunk 6 above)
Comment (edisonz0718) [reply api/cmd/main.go:46]: Yes. Will do in next batch

File: api/cmd/main.go (hunk 3 above)
Comment (edisonz0718) [reply api/cmd/main.go]: This is a fake key. Added a comment. Will remove this in the future.

File: Makefile (hunk 2 above)
Comment (edisonz0718) [reply Makefile:47]: Will do in the next batch

File: api/cmd/main.go (hunk 6 above)
Comment (edisonz0718) [reply api/cmd/main.go]: Added the config files for different envs and config flow

**Pull request merged**